libbpf-tools/funcinterval.bpf.c

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2024 Feng Yang
+// Based on funcinterval.py from BCC by Edward Wu
+
+#include "vmlinux.h"
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include "funcinterval.h"
+#include "maps.bpf.h"
+#include "bits.bpf.h"
+
+const volatile pid_t targ_pid = 0;
+const volatile bool targ_ms = false;
+
+static struct hist zero = {0};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 1);
+	__type(key, __u32);
+	__type(value, u64);
+} start SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 1);
```

Comment:
If max_entries is always one, why is a hash map needed for hists? Wouldn't an array like in funclatency.bpf.c be sufficient? 
If a single array is sufficient, several logics of this patch can be changed.
Please refer to funclatency.